cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/leveldb/LevelDBMetricsTable.java

```diff
@@ -51,8 +54,10 @@
 
   private final LevelDBTableCore core;
 
-  public LevelDBMetricsTable(String tableName, LevelDBTableService service) throws IOException {
-    this.core = new LevelDBTableCore(tableName, service);
+  public LevelDBMetricsTable(DatasetContext datasetContext, String tableName,
+                             LevelDBTableService service, CConfiguration cConf) throws IOException {
+    this.core = new LevelDBTableCore(PrefixedNamespaces.namespace(cConf, datasetContext.getNamespaceId(), tableName),
```

Comment:
may make sense to push this down so that anything that uses LevelDBTableCore must be namespace aware.